reconcile/dashdotdb_dvo.py

```diff
@@ -0,0 +1,126 @@
+        self.dashdotdb_pass = secret_content['password']
+        self.logmarker = "DDDB_DVO:"
+
+    def _post(self, deploymentvalidation):
+        if deploymentvalidation is None:
+            return None
+        cluster = deploymentvalidation['cluster']
+        dvdata = deploymentvalidation['data']
+        LOG.info('%s posting validations for %s', self.logmarker, cluster)
+        response = None
+        if not self.dry_run:
+            endpoint = (f'{self.dashdotdb_url}/api/v1/'
+                        f'deploymentvalidation/{cluster}')
+            response = requests.post(url=endpoint, json=dvdata,
+                                     auth=(self.dashdotdb_user,
+                                           self.dashdotdb_pass))
+            try:
+                response.raise_for_status()
+            except requests.exceptions.HTTPError as details:
+                LOG.error('%s error posting %s - %s',
+                          self.logmarker, cluster, details)
+
+        LOG.info('%s cluster %s synced', self.logmarker, cluster)
+        return response
+
+    def _promget(self, url, query, token=None):
+        uri = 'api/v1/query'
+        url = urljoin('https://'+url, uri)
+        headers = {
+            "accept": "application/json",
+            "Authorization": "Bearer " + token,
+        }
+        params = {'query': query}
+        LOG.info('%s Fetching prom payload from %s?%s',
+                 self.logmarker, url, query)
+        response = requests.get(url,
+                                params=params,
+                                headers=headers,
+                                timeout=(5, 30))
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as details:
+            LOG.error('%s error accessing prometheus - %s',
+                      self.logmarker, details)
+
+        response = response.json()
+        # TODO ensure len response == 1
+        # return response['data']['result']
+        return response
+
+    def _get_automationtoken(self, cluster):
+        autotoken_reader = SecretReader(settings=self.settings)
+        autotoken = {'path': "app-sre/creds/kube-configs/" + cluster,
+                     'field': "token"}
+        token = autotoken_reader.read(autotoken)
+        return token
+
+    def _get_deploymentvalidation(self, cluster, validation, oc_map):
+        LOG.debug('%s processing %s, %s', self.logmarker, cluster, validation)
+        cluster_promurl = "prometheus." + cluster + ".devshift.net"
+        promquery = "deployment_validation_"+validation+"_validation"
+        cluster_promuri = "query=" + promquery
+        cluster_promtoken = self._get_automationtoken(cluster)
+        try:
+            deploymentvalidation = self._promget(url=cluster_promurl,
+                                                 query=cluster_promuri,
+                                                 token=cluster_promtoken)
+        except StatusCodeError:
```

Comment:
I don't understand how `self._promget` is going to raise a `StatusCodeError`. You may not want to wrap `response.raise_for_status()` in line 67 in a try/except block and capture here the `requests.exceptions.HTTPError` exception

Reply:
fixed: removed try block from _promget and handling in the caller, with catchall